test/e2e/csv_e2e_test.go

```diff
@@ -312,6 +312,88 @@ func TestCreateCSVWithUnmetPermissionsCRD(t *testing.T) {
 
 }
 
+func TestCreateCSVWithNotReadyCRD(t *testing.T) {
```

Comment:
I feel like this would be easier to test with a unit test where you can manually set the status without worrying about fighting the CRD controller.